File: backend/app/schemas/audit.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class AuditLogRead(BaseModel):
    id: int
    workspace_id: int | None = None
    user_id: int | None = None
    user_display_name: str | None = None
    action: str
    resource_type: str
    resource_id: str | None = None
    detail: dict[str, Any] | None = None
    ip_address: str | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class AuditLogListResponse(BaseModel):
    items: list[AuditLogRead]
    total: int
    offset: int
    limit: int

File: frontend/src/types/audit.ts
export interface AuditLogItem {
  id: number
  workspace_id: number | null
  user_id: number | null
  user_display_name: string | null
  action: string
  resource_type: string
  resource_id: string | null
  detail: Record<string, unknown> | null
  ip_address: string | null
  created_at: string
}

export interface AuditLogListResponse {
  items: AuditLogItem[]
  total: number
  offset: number
  limit: number
}
